transport/zipkin/http_test.go

```diff
@@ -99,6 +101,59 @@ func TestHTTPOptions(t *testing.T) {
 	assert.Equal(t, "kuzhambu", sender.httpCredentials.password)
 }
 
+type recordingLogger struct {
+	errors []string
+}
+
+func (r *recordingLogger) Error(msg string) {
+	r.errors = append(r.errors, msg)
+}
+
+func (r *recordingLogger) Infof(msg string, args ...interface{}) {}
+
+func TestHTTPError(t *testing.T) {
+	_ = newHTTPServer(t)
+	sender, err := NewHTTPTransport("http://localhost:10000/api/v1/intentionally/bad/url")
+	require.NoError(t, err)
+
+	logger := new(recordingLogger)
+	tracer, closer := jaeger.NewTracer(
+		"test",
+		jaeger.NewConstSampler(true),
+		jaeger.NewRemoteReporter(sender, jaeger.ReporterOptions.Logger(logger)),
+	)
+
+	span := tracer.StartSpan("root")
+	span.Finish()
+
+	closer.Close()
+
+	require.Len(t, logger.errors, 1, "the number of errors is not 1")
+	assert.Contains(t, logger.errors[0], "error from collector: 404")
+}
+
+func TestRequestError(t *testing.T) {
```

Comment:
It seems like these two tests are only different in the URL and the error string. Might be better to do a data driven test (if you search for `t.Run` in this repo you will find many examples).